src/commands/edge/drivers/switch.ts

```diff
@@ -0,0 +1,75 @@
+import { Flags } from '@oclif/core'
+
+import { Device, DeviceIntegrationType } from '@smartthings/core-sdk'
+
+import { chooseDevice } from '@smartthings/cli-lib'
+
+import { EdgeCommand } from '../../../lib/edge-command'
+import { chooseDriver, chooseHub, listAllAvailableDrivers, listMatchingDrivers }
+	from '../../../lib/commands/drivers-util'
+
+
+export default class DriversSwitchCommand extends EdgeCommand<typeof DriversSwitchCommand.flags> {
+	static description = 'change the driver used by an installed device'
+
+	static examples = [`# switch driver, prompting user for all necessary input
+$ smartthings edge:drivers:switch
+
+# switch driver, including all necessary input on the command line
+$ smartthings edge:drivers:switch --hub <hub-id> --driver <driver-id> <device-id>`,
+	`
+# include all available drivers in prompt, even if they don't match the chosen device
+$ smartthings edge:drivers:switch --include-non-matching`,
+	]
+
+	static flags = {
+		...EdgeCommand.flags,
+		hub: Flags.string({
+			char: 'H',
+			description: 'hub id',
+		}),
+		driver: Flags.string({
+			char: 'd',
+			description: 'id of new driver to use',
+		}),
+		// eslint-disable-next-line @typescript-eslint/naming-convention
+		'include-non-matching': Flags.boolean({
+			char: 'I',
+			description: 'when presenting a list of drivers to switch to, include drivers that do not match the device',
+		}),
+	}
+
+	static args = [{
+		name: 'deviceId',
+		description: 'id of device to update',
+	}]
+
+	async run(): Promise<void> {
+		const hubId = await chooseHub(this, 'Which hub is the device you want to switch using?',
```

Comment:
This took me a few reads to fully understand. Since switch is already in the context of the command being used, we could shorten to something like `Which hub is the device connected to?`.

Reply:
I like your text better too. Fixed.